### ISSUE TITLE
remove publish attribute from actions

### ISSUE THREAD

**Issue by bjustin-ibm** [openwhisk/openwhisk#1359]
After some experimentation, it became clear that the `publish` attribute for actions is not needed and, worse, can lead to confusion.

For example, create a package with `--shared yes` then create an action in that package also with `--shared yes`. Running `wsk package get` shows that the action has a `publish` value of `false`.  However, running `wsk action get` on the actions shows the `publish` value set to `true`.

Additionally, running `wsk action create --shared no` to create an action inside a published package does not prevent others from invoking the action - that is, the `publish` value has no effect.

Lastly... creating an action outside of a shared package with `--shared yes` does not allow others to invoke that action.

*tl;dr* The `publish` attribute for actions doesn't do anything, and should be removed.

This will affect the backend, CLI, and documentation.

**Comment by bjustin-ibm** [comment]
Mostly a duplicate of #465
